core/templates/components/summary-tile/topic-summary-tile.component.html

```diff
@@ -11,7 +11,12 @@
       </picture>
     </div>
     <div class="topic-details" [ngStyle]="{'background-color': getDarkerThumbnailBgColor()}" ng-class="{'mask-effect': !isPublished}">
-      <span class="topic-name">{{ topicSummary.getName() }}</span>
+      <span class="topic-name" *ngIf="!topicNameTranslationKeyIsToBeDisplayed">
```

Comment:
I think topicNameTranslationKeyIsDisplayed is simpler to read.

However, the name seems wrong. You're not displaying the key, right -- you're displaying its translation? So what you've currently named it as seems a bit misleading.

Maybe something like isHackyTranslationDisplayed / hackyTopicNameTranslationIsDisplayed / etc.? (please modify as needed)

Reply:
Done!